sklearn/metrics/_regression.py

```diff
@@ -387,7 +395,12 @@ def median_absolute_error(y_true, y_pred, *, multioutput='uniform_average'):
     """
     y_type, y_true, y_pred, multioutput = _check_reg_targets(
         y_true, y_pred, multioutput)
-    output_errors = np.median(np.abs(y_pred - y_true), axis=0)
+    sample_weight = _check_sample_weight(sample_weight, y_pred)
+    if sample_weight is None:
+        output_errors = np.median(np.abs(y_pred - y_true), axis=0)
+    else:
+        output_errors = _weighted_percentile(np.abs(y_pred - y_true),
```

Comment:
```suggestion
        sample_weight = _check_sample_weight(sample_weight, y_pred)
        output_errors = _weighted_percentile(np.abs(y_pred - y_true),
```

Reply:
You need to add the checking here otherwise we will create an array of 1s.
Then we will never compute the median with NumPy.

Reply:
(This something which is written in the docstring of the `_check_sample_weight` but it could be tricky to find out :)